src/content/datachannel/messaging/main.js

```diff
@@ -0,0 +1,180 @@
+    this.connected = false;
+    // this.shadowRoot.querySelector('#sendLocal').disabled = true;
+    // this.shadowRoot.querySelector('#sendRemote').disabled = true;
+    // this.shadowRoot.querySelector('#connectButton').disabled = false;
+  }
+
+  _onLocalMessageReceived(event) {
+    console.log(`Remote message received by local: ${event.data}`);
+    this.shadowRoot.querySelector('#localIncoming').value += event.data + '\n';
+  }
+
+  _onRemoteDataChannel(event) {
+    console.log(`onRemoteDataChannel: ${JSON.stringify(event)}`);
+    this._remoteChannel = event.channel;
+    this._remoteChannel.binaryType = 'arraybuffer';
+    this._remoteChannel.addEventListener('message', e => this._onRemoteMessageReceived(e));
+    this._remoteChannel.addEventListener('close', e => this._channelClosed(e));
+  }
+
+  _onRemoteMessageReceived(event) {
+    console.log(`Local message received by remote: ${event.data}`);
+    this.shadowRoot.querySelector('#remoteIncoming').value += event.data + '\n';
+  }
+
+  static get properties() {
+    return {
+      connected: {
+        type: Boolean,
+        notify: true,
+        reflectToAttribute: true
+      }
+    };
+  }
+
+  _render({connected}) {
+    return html`
+    <section>
+        <style>
+        @import "../../../css/main.css";
+        @import "main.css";
+        </style>
+        <div>
+            <button disabled="${connected}" id="connectButton">Connect</button>
+            <button disabled="${!connected}" id="disconnectButton">Disconnect</button>
+        </div>
+
+        <div class="messageBox">
+            <label for="localOutgoing">Local outgoing message:</label>
+            <textarea class="message" id="localOutgoing" 
+                      placeholder="Local outgoing message goes here."></textarea>
+            <button disabled="${!connected}" id="sendLocal">Send message from local</button>
+        </div>
+        <div class="messageBox">
+            <label for="localIncoming">Local incoming messages:</label>
+            <textarea class="message" id="localIncoming" disabled 
+                      placeholder="Local incoming messages arrive here."></textarea>
+        </div>
+
+        <div class="messageBox">
+            <label for="remoteOutgoing">Remote outgoing message:</label>
+            <textarea class="message" id="remoteOutgoing" 
+                      placeholder="Remote outgoing message goes here."></textarea>
+            <button disabled="${!connected}" id="sendRemote">Send message from remote</button>
+        </div>
+        <div class="messageBox">
+            <label for="remoteIncoming">Remote incoming messages:</label>
+            <textarea class="message" id="remoteIncoming" disabled
+                      placeholder="Remote incoming messages arrive here."></textarea>
+        </div>
+
+    </section>`;
+  }
+
+  _firstRendered() {
+    this.shadowRoot.querySelector('#connectButton').addEventListener('click', () => this.connect());
```

Comment:
You can add these in the HTML.

`<button disabled="${connected}" @click={() => this.connect()}>Connect</button>`

Reply:
actually you probably want

`<button ?disabled=${connected} @click={() => this.connect()}>Connect</button>`